README.md

```diff
@@ -156,17 +160,78 @@ function Tickler() {
 ```
 
 Just as in JavaScript, when we handle events in React, we can provide an `event`
-parameter to our event handler callback function. When the button is clicked, we
-can access all the information about the event (the event `target`, mouse
-coordinates via `clientX` and `clientY`, etc).
+parameter to our event handler callback function. Now that we're using TypeScript,
+however, we also have to type our `event` so TypeScript knows what kind of event 
+the function should expect. Let's take a moment to discuss typing events before 
+we get back to the `onClick`.
+
+### Event Types
+ 
+When using React, we can take advantage of the type interfaces they provide. In
+this particular case, for example, we're using the `React.MouseEvent` interface. 
+This type is used for, well, mouse events such as `onClick`, `onMouseEnter`, 
+`onDoubleClick` and plenty more. 
+
+All events fall under a specific umbrella, one being the aforementioned `MouseEvent`. 
+We'll learn more as we go along, but one important thing to note is that this is where 
+your IDE can come in handy. Assuming your IDE supports TypeScript, hovering over an
+event attribute will show you the umbrella type its return belongs to. For example: 
+
+![Type finding demonstration in VSCode](assets/type-finding.gif)
+
+Here, VSCode tells us that the `onClick` attribute expects a `React.MouseEventHandler` 
+to be passed to it. By extrapolation, this then tells us the event that `onClick` will
+return to the handler is of type `React.MouseEvent`. 
+
+However, typing doesn't stop there.
+
+When we're dealing with events, we can put them on virutally any DOM element we want. 
+In our `onClick` example, we put it on a `button` which is an `HTMLButtonElement` type. 
+But we could have put it on a `div` which is an `HTMLDivElement` type. This means 
+`React.MouseEvent` can't specify the exact element type the event came from. 
+
+This is where the `<>` syntax comes in. You may recall, this syntax denotes a 
+[**generic**][generics] type. When paired with an interface as in our examples here, 
+it lets us type our events with the interface on any type of element. We just 
```

Comment:
This use of "interface" is a little confusing - you may want to introduce it earlier in the section